backend/internal/user/service/userservice.go

```diff
@@ -294,6 +325,30 @@ func (as *UserService) UpdateUser(userID string, user *model.User) (*model.User,
 		return nil, &constants.ErrorInvalidRequestFormat
 	}
 
+	if svcErr := as.userSchemaService.ValidateUser(user.Type, user.Attributes); svcErr != nil {
+		return nil, svcErr
+	}
+
+	isValid, svcErr := as.userSchemaService.ValidateUserUniqueness(user.Type, user.Attributes,
+		func(filters map[string]interface{}) (*string, error) {
+			userID, svcErr := as.IdentifyUser(filters)
+			if svcErr != nil {
+				if svcErr.Code == constants.ErrorUserNotFound.Code {
+					return nil, nil
+				} else {
+					return nil, errors.New(svcErr.Error)
+				}
+			}
+			return userID, nil
+		})
+	if svcErr != nil {
+		return nil, svcErr
+	}
```

Comment:
Shall we extract this validation code block into a helper method as this is repeated in `CreateUser` as well?